### PR TITLE
Activity & Conversation new filters

### PR DESCRIPTION
**Pull request by gaspergrom**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at 3f1ad74</samp>

This pull request refactors the activity and conversation modules to use the pinia store and the CrFilter component for data and filter management, and to improve the UI and functionality of the filter components. It also fixes a typo in the auth layout and adds a search filter for the activity list. The changes affect the following files: `activity-list.vue`, `ActivityTypeFilter.vue`, `config.ts` for various filters, `activity-list-page.vue`, `conversation-list.vue`, `activity-service.js`, `actions.ts`, `getters.ts`, `index.ts`, `state.ts`, `auth-layout-side.vue`, `ChannelFilter.vue`, and `config.ts` for the channel filter.
​

### <samp>🤖 Generated by Copilot at 3f1ad74</samp>

> _Sing, O Muse, of the mighty deeds of the code reviewers,_
> _who refactored and improved the activity and conversation modules,_
> _with pinia stores and CrFilters, `defineStore` and `itemLabelRenderer`,_
> _and made the UI more elegant and responsive, like the work of Hephaestus._

### <samp>🤖 Generated by Copilot at 3f1ad74</samp>

*  Add a CrFilter component to the activity-list.vue and conversation-list.vue components, which allows the user to apply filters to the activity or conversation list using various filter types and options ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-1030c34aa088f6717a19a05ec29ba8f7b79ee813058e79944c13ea97e700655dR3-R9),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-b7041f192c765f4fd0909a00efc600db5ef3581f3fae47f95b322fff5dd21ca6R3-R9))
*  Remove the app-pagination-sorter component from the activity-list.vue and conversation-list.vue components, as the CrFilter component handles the pagination and sorting logic internally ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-1030c34aa088f6717a19a05ec29ba8f7b79ee813058e79944c13ea97e700655dL22-R31),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-b7041f192c765f4fd0909a00efc600db5ef3581f3fae47f95b322fff5dd21ca6L19-R35))
*  Replace the useStore hook with the useActivityStore and useConversationStore hooks in the activity-list.vue and conversation-list.vue components, which are pinia stores that manage the activity and conversation state and actions ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-1030c34aa088f6717a19a05ec29ba8f7b79ee813058e79944c13ea97e700655dL78-R96),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-b7041f192c765f4fd0909a00efc600db5ef3581f3fae47f95b322fff5dd21ca6L69-R85))
*  Remove some unused props and data properties from the activity-list.vue and conversation-list.vue components, as they are now managed by the pinia stores or the CrFilter component ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-1030c34aa088f6717a19a05ec29ba8f7b79ee813058e79944c13ea97e700655dL105-R152),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-b7041f192c765f4fd0909a00efc600db5ef3581f3fae47f95b322fff5dd21ca6L93-R130))
*  Define the pinia activity and conversation stores, which are reactive and type-safe stores that contain the state, getters, and actions for the activity and conversation modules ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-65d4191552645d95e79c81e22bac914a0e7499391cb8da06aa7baf686112d9e8R1-R21),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-33ef23d8e6c9709404519591c7ba841c6ebc53f8794b6c382549fb023f8988ffR1-R2),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-7ed78b5af4ac9c23a72c4bf09004aac3b199c1516c925b0f649278cc6aa1f672R1-R13),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-857f6fcf44cb71b4814635029c9f44543c515fb116ebac9865eeadc37d1459abR1-R26))
*  Add a fetchActivities action to the pinia activity store, which makes an API call to list activities based on a given body parameter and updates the state with the response data ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-65d4191552645d95e79c81e22bac914a0e7499391cb8da06aa7baf686112d9e8R1-R21))
*  Add a new static method to the ActivityService class that makes an API call to list activities based on a given body parameter and returns the response data ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-39013f6997a8249afda3999ea16912935cb9531e48c8c8164a9e8c24846dd463R124-R142))
*  Replace the app-activity-list-tabs component with an el-tabs component in the activity-list-page.vue component, which renders a tabbed navigation for the activity and conversation views ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-0ce6547bee0155186a048d673be3dafaab869e51b92902f7c48d524d07467c58L31-R48))
*  Replace the script tag with a script setup tag in the activity-list-page.vue component, which allows defining the component options using the Composition API and TypeScript ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-0ce6547bee0155186a048d673be3dafaab869e51b92902f7c48d524d07467c58L62-R69))
*  Remove some unused props and data properties from the activity-list-page.vue component, and define some refs and a function that handle the activity drawer logic and the tab change logic ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-0ce6547bee0155186a048d673be3dafaab869e51b92902f7c48d524d07467c58L71-R104))
*  Add a new file that defines the channel filter component for the conversation list, which is a custom select filter component that allows the user to filter conversations by the channel name ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-b5b956b1907bbed853c9153b779779e3b2950fa52907d5ef27af7f5b5956c18fR1-R49))
*  Add a new file that defines the channel filter config for the conversation list, which is a custom filter config object that contains the configuration for the channel filter component ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-218f115f0e298336bb60b737c1cd4272f87261c8f5213c7780f745d35ea1f111R1-R30))
*  Modify the options, itemLabelRenderer, and apiFilterRenderer properties of the platform, member, date, and sentiment filter configs, to use the enabledConfigs property of the CrowdIntegrations object, the itemLabelRendererByType function, and the value and include properties of the multi-select and select filters ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-cd8e780c8488cffad4194da57b9385504fa0d90b0b7084b6523dc9a05795a651L4-R6),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-cd8e780c8488cffad4194da57b9385504fa0d90b0b7084b6523dc9a05795a651L16-R32),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-a9caf8fef766c417b77c51900f046e8bbab2f34427bd53ac5a3ee24fa50febe8L3-R6),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-a9caf8fef766c417b77c51900f046e8bbab2f34427bd53ac5a3ee24fa50febe8L15-R26),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-000975af376ec51052285dec3052d0d2b7b3b5bb5a32c937973df3719fab0af7L2-R7),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-000975af376ec51052285dec3052d0d2b7b3b5bb5a32c937973df3719fab0af7L13-R27),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-e2ef4c4cfd836c42d5d9305bfb8920729b1ac2076a16d0d9a1d431429732b038L3),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-e2ef4c4cfd836c42d5d9305bfb8920729b1ac2076a16d0d9a1d431429732b038L32-R37),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-acaec04bcea58bbc89fb09cdcf9c13486baeb8031fbf36443bdf9a63194cee40L2-R8),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-acaec04bcea58bbc89fb09cdcf9c13486baeb8031fbf36443bdf9a63194cee40L14-R19),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-acaec04bcea58bbc89fb09cdcf9c13486baeb8031fbf36443bdf9a63194cee40L26-R40),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-3a3c6c5d10f2d009d72f8eba26e51c75b589b33911727ad0e566568b574e201aL2-R7),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-3a3c6c5d10f2d009d72f8eba26e51c75b589b33911727ad0e566568b574e201aL13-R31))
*  Modify the options and data props of the activityType and channel filter components, to use the types property of the activityTypeStore and the currentTenant getter from the vuex store, and emit the update:data event when changed ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-f7e958bff5ca82a24a2bcc809b0d5868c76f6c7378c61dcb0b8bd287a646edd2L2-R26),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-f7e958bff5ca82a24a2bcc809b0d5868c76f6c7378c61dcb0b8bd287a646edd2L21-R49),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-69b2549353f117ee8aa475b258a2b7e5c4d1dabf070389e6fb0cc4515961bf2bL2-R27),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-69b2549353f117ee8aa475b258a2b7e5c4d1dabf070389e6fb0cc4515961bf2bL21-R48))
*  Remove the optional chaining operator from the key attribute of the app-activity-item component, as the activity.id should always be defined and not nullish ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-1030c34aa088f6717a19a05ec29ba8f7b79ee813058e79944c13ea97e700655dL35-R42))
*  Add a new export to the main filter config file, which is the activitySearchFilter object that contains the configuration for the search filter component for the activity list ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-23ca53439be6cf4f27ec40bc2c06f04ea84246c16d275d2f87590ec836448089R2),[link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-23ca53439be6cf4f27ec40bc2c06f04ea84246c16d275d2f87590ec836448089R18-R30))
*  Fix a typo in the auth-layout-side.vue component, which adds a missing comma after the word 'product' in the paragraph text ([link](https://github.com/CrowdDotDev/crowd.dev/pull/955/files?diff=unified&w=0#diff-4b2cbb3daa9ca91bea3b7d6a5b35ebe659cd39ad83258830d6fab4fe58dc4401L16-R16))

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screehshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: frontend/src/modules/activity/config/filters/main.ts [hunk 1]
```diff
@@ -14,3 +15,16 @@ export const activityFilters: Record<string, FilterConfig> = {
...
   sentiment,
 };
+export const activitySearchFilter: SearchFilterConfig = {
+  placeholder: 'Search activities',
+  apiFilterRenderer(value: string): any[] {
+    return [
+      {
+        or: [
+          { title: { textContains: value } },
+          { body: { contains: [value] } },
```
Comment (joanagmaia) [inline frontend/src/modules/activity/config/filters/main.ts:25]: Should `body` also be `textContains` with string? At least that's how we are using it right now and it's a string value

File: frontend/src/modules/conversation/components/conversation-list.vue [hunk 2]
```diff
@@ -16,16 +23,16 @@
...
+          <!--          <app-pagination-sorter-->
+          <!--            v-model="sorterFilter"-->
+          <!--            :page-size="Number(pagination.pageSize)"-->
+          <!--            :total="count"-->
+          <!--            :current-page="pagination.currentPage"-->
+          <!--            :has-page-counter="false"-->
+          <!--            module="conversation"-->
+          <!--            position="top"-->
+          <!--            @change-sorter="doChangeFilter"-->
+          <!--          />-->
```
Comment (joanagmaia) [inline frontend/src/modules/conversation/components/conversation-list.vue:35]: Checking if this is supposed to be commented

**Review by joanagmaia — COMMENTED** [review]
Looks great!
One thing I noticed while testing was that conversation query endpoint is returning 500. Looked at the logs and it seems that is because of the `orderBy` value being passed:
`column conversation.timestamp does not exist`

File: frontend/src/modules/activity/config/filters/main.ts (hunk 1 above)
Comment (gaspergrom) [reply frontend/src/modules/activity/config/filters/main.ts:25]: fixed

**Pull request merged**